### ISSUE TITLE
the loss of diffusion model is wrong in step1_train_svd.py

### ISSUE THREAD

**Issue by cjmdd** [roboterax/video-prediction-policy#4]
model_pred = unet(input_latents, c_noise, encoder_hidden_states=encoder_hidden_states, added_time_ids=added_time_ids).sample
predict_x0 = c_out * model_pred + c_skip * noisy_latents 
loss += ((predict_x0 - latents)**2 * loss_weight).mean()

This formula, c_out * model_pred + c_skip * noisy_latents  , was originally used to restore latents from noise predictions, rather than defining the objective of the loss function. Model_pred is already the predicted value of latents, so there is no need to use the c_out and c_skip transformations again.

Moreover, there are some incorrect loss:

    def loss(self, state, action, goal, noise, sigma, **kwargs):
        """
        Compute the loss for the denoising process.

        Args:
            state: The input state.
            action: The input action.
            goal: The input goal.
            noise: The input noise.
            sigma: The input sigma.
            **kwargs: Additional keyword arguments.
        Returns:
            The computed loss.
        """
        c_skip, c_out, c_in = [append_dims(x, action.ndim) for x in self.get_scalings(sigma)]
        noised_input = action + noise * append_dims(sigma, action.ndim)
        model_output = self.inner_model(state, noised_input * c_in, goal, sigma, **kwargs)
        target = (action - c_skip * noised_input) / c_out
        return (model_output - target).pow(2).flatten(1).mean(), model_output

    def diffusion_loss(
            self,
            perceptual_emb: torch.Tensor,
            latent_goal: torch.Tensor,
            actions: torch.Tensor,
    ) -> torch.Tensor:
        """
        Computes the score matching loss given the perceptual embedding, latent goal, and desired actions.
        """
        self.model.train()
        sigmas = self.make_sample_density()(shape=(len(actions),), device=self.device).to(self.device)
        noise = torch.randn_like(actions).to(self.device)
        target, model_output = self.model.loss(perceptual_emb, actions, latent_goal, noise, sigmas)
        return target, model_output

where target=(model_output - target).pow(2).flatten(1).mean() is wrong, the target should be action.

target, model_output = self.diffusion_loss(
            predictive_feature,
            latent_goal,
            dataset_batch["actions"],
        )
        loss_dict = {}
        loss_dict['loss_xyz'] = (model_output[:,:,:3] - target[:,:,:3]).pow(2).mean()*3/38+(model_output[:,:,19:22] - target[:,:,19:22]).pow(2).mean()*3/38

**Comment by Robert-gyj** [comment]
Thank you for the question. I think maybe you misunderstand the loss equations.

According to the original SVD paper page 18 (https://arxiv.org/pdf/2311.15127), they implemented diffusion process through the numerical simulation of the **Probability Flow ODE**, so the loss equation is different from original DDPM (which predict x_0). You can read the original SVD paper for more details.
